lib/perl/Genome/Model/Command/Admin/ModelSummary.pm

```diff
@@ -236,18 +244,18 @@ sub latest_build_succeeded {
 }
 
 
-my $max_fails = 5;
-sub model_has_failed_to_many_times {
+sub model_has_failed_too_many_times {
+    my $self = shift;
     my $model = shift;
 
-    my @builds = reverse $model->builds;
-    return unless @builds;
+    my $max_fails = $self->max_fail_count;
+    return unless defined $max_fails;
 
-    my $n = (@builds >= $max_fails ? ($max_fails - 1) : $#builds);
-    my @last_n_builds = @builds[0..$n];
+    my @builds = $model->builds;
+    return unless @builds;
 
-    my @failed_builds = grep { $_->status eq 'Failed' } @last_n_builds;
-    return (@failed_builds == $max_fails );
+    my @failed_builds = grep { $_->status eq 'Failed' } @builds;
```

Comment:
ModelSummary currently pre-loads all the builds into memory.  After this and #765 I'd like to try to make many changes to be more careful about loading.  I'll make sure taking out this `grep` is one of those changes!